### ISSUE TITLE
youtubeSignupRequired /  Unauthorized error

### ISSUE THREAD

**Issue by shiffman** [IonicaBizau/youtube-api#36]
Thanks for this incredibly helpful node module.  I've followed your `JWT` tutorial and believe I'm authenticating correctly.  However, I'm getting the following errors:

```
{ [Error: Unauthorized]
  code: 401,
  errors: 
   [ { domain: 'youtube.header',
       reason: 'youtubeSignupRequired',
       message: 'Unauthorized',
       locationType: 'header',
       location: 'Authorization' } ] }
```

I'm using the youtube upload example referenced in #20.  My JWT authorization looks like:

```javascript
Youtube.authenticate({
  "type": 'jwt',
  "email":    ".........@developer.gserviceaccount.com"
  , "keyFile":  "./keyFile.pem"
  , "key": "-----BEGIN PRIVATE KEY-----\n.....\n-----END PRIVATE KEY-----\n"
  , "scopes":   ["https://www.googleapis.com/auth/youtube"]
});
```

Did I miss a step perhaps?

**Comment by IonicaBizau** [comment]
I will try to add a little example soon (next hours). I used JWT in one project, and I will check there how JWT works.

**Comment by shiffman** [comment]
Thanks!  I am not certain JWT is the best path for me.  I am working on an example that auto-upload videos *only* from my account via server instance.  So authenticating a different user is unnecessary and no client-side code needed. (at the moment.)   I will keep investigating.

**Comment by IonicaBizau** [comment]
> So authenticating a different user is unnecessary and no client-side code needed (at the moment).

OK, you do need JWT! :smile:

**Comment by shiffman** [comment]
Ok, good to know I am on the right path!  If I can get it working I'll be glad to share an example with a pull request.

**Comment by shiffman** [comment]
Additional info, if I add:

```javascript

Youtube.authenticate({
   // etc.
  , subject: "my.mail@gmail.com" // optional
  // etc.
});
```
I get a different error:

```
[Error: unauthorized_client]
```

**Comment by IonicaBizau** [comment]
The bad news is that JWT cannot access YouTube resources. The good news is that I already have an example which almost works with OAuth2. :smile:

**Comment by IonicaBizau** [comment]
@shiffman OK, so I addressed this in #37. :boom:

**Comment by shiffman** [comment]
Fantastic, taking a look now and will resume any discussion there!
